crates/circuit/src/dag_circuit.rs

```diff
@@ -7047,6 +6846,273 @@ impl DAGCircuit {
+                            let py_op = op.instruction.bind(py);
+                            let py_op = py_op.call_method0(intern!(py, "to_mutable"))?;
+                            if py_op.is_instance(imports::IF_ELSE_OP.get_bound(py))?
+                                || py_op.is_instance(imports::WHILE_LOOP_OP.get_bound(py))?
+                            {
+                                if let Ok(condition) = py_op.getattr(intern!(py, "condition")) {
+                                    match variable_mapper {
+                                        Some(ref variable_mapper) => {
+                                            let condition =
+                                                variable_mapper.map_condition(&condition, true)?;
+                                            py_op.setattr(intern!(py, "condition"), condition)?;
+                                        }
+                                        None => {
+                                            let var_mapper = build_var_mapper(&self.cregs)?;
+                                            let condition =
+                                                var_mapper.map_condition(&condition, true)?;
+                                            py_op.setattr(intern!(py, "condition"), condition)?;
+                                            variable_mapper = Some(var_mapper);
+                                        }
+                                    }
+                                }
+                            } else if py_op.is_instance(imports::SWITCH_CASE_OP.get_bound(py))? {
+                                match variable_mapper {
+                                    Some(ref variable_mapper) => {
+                                        py_op.setattr(
+                                            intern!(py, "target"),
+                                            variable_mapper.map_target(
+                                                &py_op.getattr(intern!(py, "target"))?,
+                                            )?,
+                                        )?;
+                                    }
+                                    None => {
+                                        let var_mapper = build_var_mapper(&self.cregs)?;
+                                        py_op.setattr(
+                                            intern!(py, "target"),
+                                            var_mapper.map_target(
+                                                &py_op.getattr(intern!(py, "target"))?,
+                                            )?,
+                                        )?;
+                                        variable_mapper = Some(var_mapper);
+                                    }
+                                }
+                            }
+                            PackedInstruction {
+                                op: PackedOperation::from_instruction(Box::new(PyInstruction {
```

Comment:
I could be wrong, but we might implement `From<PyInstruction>` for `PackedOperation` to automatically move it into a `Box` and give you what you want.

Reply:
Oh yeah, I forgot about that I can replace `Box::new()` with `.into()`. I tried it without the explicit box and the compiler wasn't able to do it automatically.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14164/commits/c6e988eb6fb4420f96bbd03a78e44583ef4a560d